clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -8345,6 +8345,8 @@ def err_cast_to_bfloat16 : Error<"cannot type-cast to __bf16">;
 def err_cast_from_bfloat16 : Error<"cannot type-cast from __bf16">;
 def err_typecheck_expect_scalar_operand : Error<
   "operand of type %0 where arithmetic or pointer type is required">;
+def err_typecheck_expect_capability_operand : Error<
+  "operand of type %0 where capability pointer or (u)intcap is required">;
```

Comment:
I'd be inclined to just say "... where capability is required". But if you want the explicit "pointer or (u)intcap" then we should at least have the _t suffix.

Reply:
Yeah I was trying to imply that references aren't okay even though they are capabilities.

Reply:
Oh, hm, well I think that's a given, references behave like their referred-to type, not a pointer, even aside from CHERI. And references _are_ ok, so long as they're references to capabilities?

Reply:
I'll change it to capability type.